### PR TITLE
`batch_mul_with_preprocessing` no longer takes `self` as argument

### PR DESCRIPTION
**Pull request by mmagician**
## Description

Keeping up-to-date with https://github.com/arkworks-rs/algebra/pull/757

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: poly-commit/src/marlin/marlin_pst13_pc/mod.rs [hunk 1]
```diff
@@ -236,7 +236,7 @@ where
                     cur *= &betas[i];
                     powers_of_beta.push(cur);
                 }
-                *v = gamma_g.batch_mul_with_preprocessing(&powers_of_beta, &gamma_g_table);
+                *v = E::G1::batch_mul_with_preprocessing(&gamma_g_table, &powers_of_beta);
```
Comment (Pratyush) [inline poly-commit/src/marlin/marlin_pst13_pc/mod.rs:239]:
```suggestion
                *v = gamma_g1_table.batch_mul_with_preprocessing(&powers_of_beta);
```

File: poly-commit/src/multilinear_pc/mod.rs [hunk 2]
```diff
@@ -59,7 +59,7 @@ impl<E: Pairing> MultilinearPC<E> {
         }
 
         let g_table = BatchMulPreprocessing::new(g, num_vars);
-        let pp_g = g.batch_mul_with_preprocessing(&pp_powers, &g_table);
+        let pp_g = E::G1::batch_mul_with_preprocessing(&g_table, &pp_powers);
```
Comment (Pratyush) [inline poly-commit/src/multilinear_pc/mod.rs:62]:
```suggestion
        let pp_g = g_table.batch_mul_with_preprocessing(&pp_powers);
```

File: poly-commit/src/multilinear_pc/mod.rs [hunk 3]
```diff
@@ -73,7 +73,7 @@ impl<E: Pairing> MultilinearPC<E> {
 
         // uncomment to measure the time for calculating vp
         // let vp_generation_timer = start_timer!(|| "VP generation");
-        let g_mask = g.batch_mul_with_preprocessing(&t, &g_table);
+        let g_mask = E::G1::batch_mul_with_preprocessing(&g_table, &t);
```
Comment (Pratyush) [inline poly-commit/src/multilinear_pc/mod.rs:76]:
```suggestion
        let g_mask = g_table.batch_mul(&t);
```

**Comment by mmagician** [comment]
@Pratyush I re-ran CI and indeed, all is well

**Review by Pratyush — APPROVED** [review]

**Pull request merged**